internal/controller/ols_app_server_reconciliator.go

```diff
@@ -0,0 +1,152 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+)
+
+func (r *OLSConfigReconciler) reconcileAppServer(ctx context.Context, olsconfig *olsv1alpha1.OLSConfig) error {
+	r.logger.Info("reconcileAppServer starts")
+	tasks := []ReconcileTask{
+		{
+			Name: "reconcile ServiceAccount",
+			Task: r.reconcileServiceAccount,
+		},
+		{
+			Name: "reconcile OLSConfigMap",
+			Task: r.reconcileOLSConfigMap,
+		},
+		{
+			Name: "reconcile Deployment",
+			Task: r.reconcileDeployment,
+		},
+		{
+			Name: "reconcile Service",
+			Task: r.reconcileService,
+		},
+	}
+
+	for _, task := range tasks {
+		err := task.Task(ctx, olsconfig)
+		if err != nil {
+			return fmt.Errorf("failed to %s: %w", task.Name, err)
+		}
+	}
+
+	r.logger.Info("reconcileAppServer completes")
+
+	return nil
+}
+
+func (r *OLSConfigReconciler) reconcileOLSConfigMap(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
+	cm, err := r.generateOLSConfigMap(cr)
+	if err != nil {
+		return fmt.Errorf("failed to generate OLS configmap: %w", err)
+	}
+
+	foundCm := &corev1.ConfigMap{}
+	err = r.Client.Get(ctx, client.ObjectKey{Name: OLSConfigCmName, Namespace: OLSNamespaceDefault}, foundCm)
+	if err != nil && errors.IsNotFound(err) {
+		err = r.Create(ctx, cm)
+		if err != nil {
+			return fmt.Errorf("failed to create OLS configmap: %w", err)
+		}
+	} else if err != nil {
```

Comment:
Can we log the error here ?

Reply:
More information is added to the error message. The message is ouput into the log at line 40.
```go
for _, task := range tasks {
		err := task.Task(ctx, olsconfig)
		if err != nil {
			return fmt.Errorf("failed to %s: %w", task.Name, err)
		}
	}
```